extensions/Maps/includes/Maps_Geocoder.php

```diff
@@ -111,7 +111,21 @@ public function hasAlias( $alias ) {
 	 * @return array or false
 	 */
 	public function geocode( $address ) {
-		$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+
+		// MAIN-6654: cache requests to this API because it has daily limits
+		global $wgMemc;
+		$key = wfMemcKey( "geocoder", $address );
+		$response = $wgMemc->get( $key );
+		if ( empty($response) ) {
+			$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+			// MAIN-6654: make sure we do not return or store a bad response because it causes errors later
+			if ( substr( $response, "OVER_QUERY_LIMIT" ) !== false ) {
```

Comment:
Nice catch.  I tested it on sandbox and then used the wrong function when copying over.   Re-testing and updating the PR now.  

As far as the caching strategy, I realized that caching for 1 day is pointless since there's a daily API limit so I increased the duration to 30 days, and we probably don't want to cache a bad response that long.